crates/ty_python_semantic/resources/mdtest/generics/legacy/functions.md

```diff
@@ -379,14 +379,13 @@ T = TypeVar("T")
 def invoke(fn: Callable[[A], B], value: A) -> B:
     return fn(value)
 
-def identity(x: T) -> T:
+def identity(x: T, /) -> T:
```

Comment:
This is an interesting nuance: `Callable[[A], B]` creates a signature containing positional-only parameters, so we have to make the signature here positional-only as well to make `identity` pass the assignability check for it to be a valid argument to the `fn` parameter.

(That said, I'm not convinced that's...correct? Are we mixing up the ordering of the assignability check operands somewhere?)

Reply:
Ok, I looked into this a bit more and what I'm seeing is that:

In `infer_map_impl`, the formal signature is `Callable[[A], B]` and the actual signature is the `identity` function and you invoke `formal_signature.when_constraint_set_assignable_to(..., actual_signature, ...)` which means we're checking when is `Callable[[A], B]` assignable to `identity` function which leads to checking when is the positional-only parameter (`A`) is assignable to positional-or-keyword parameter (`a: int`) which is never. This is because you cannot substitute a callable that takes a positional-or-keyword parameter with a callable that takes a positional-only parameter (https://play.ty.dev/7b31e7a0-19c8-4d2b-8bf6-507764372c0d).

Reply:
It looks like this comment still needs follow-up? Why are we checking assignability in the wrong direction (requiring the formal signature to be assignable to the actual signature)? I don't think the change here and below to make `identity` and `head` take a positional-only parameter should be required.